### PR TITLE
Setup typechecking and dev mode

### PR DESCRIPTION
**Pull request by damianpm**
Fixes https://github.com/3scale-labs/kamrad/issues/2

- Set up type checking
- Adds "lint" and "dev" npm scripts
- Fixes some type errors

**Typechecking:**
```
npm run lint
```

**Dev mode:**
```
npm run dev
```

### CODE REVIEW

File: assets/js/greeter/Greeter.tsx [hunk 1]
```diff
@@ -1,14 +1,14 @@
 import React, { useState, useEffect } from 'react'
 
 interface Props {
-	id: number
```
Comment (didierofrivia) [inline assets/js/greeter/Greeter.tsx:4]: let's keep the 2 spaces, either way the nesting will be insane

File: package.json [hunk 2]
```diff
@@ -4,6 +4,8 @@
   "description": "Kuadrant's developer portal",
   "main": "src/index.ts",
   "scripts": {
+    "dev": "concurrently \"npm run lint\" \"hugo server\"",
```
Comment (didierofrivia) [inline package.json:7]: we will keep all scripts in a Makefile, but I think this doesn't hurt right now

File: assets/js/greeter/Greeter.tsx (hunk 1 above)
Comment (damianpm) [reply assets/js/greeter/Greeter.tsx:4]: I'm using 2 spaces tab for indentation, same as `index.ts`

File: package.json [hunk 3]
```diff
@@ -34,7 +36,7 @@
     "@types/react": "^17.0.5",
     "@types/react-dom": "^17.0.4",
     "babel-loader": "^8.2.2",
-    "npm-run-all": "^1.7.0",
+    "concurrently": "^6.2.0",
```
Comment (didierofrivia) [inline package.json:39]: why change to concurrently when `npm-run-all` was chosen?

File: assets/js/greeter/Greeter.tsx (hunk 1 above)
Comment (didierofrivia) [reply assets/js/greeter/Greeter.tsx:4]: Then why it's triggering those changes?

File: config.toml [hunk 4]
```diff
@@ -46,3 +46,8 @@ defaultContentLanguageInSubdir = false
 title = "Kamrad"
 languageName = "English"
 weight = 1
+
+# Build
+[build]
+  noJSConfigInAssets = true
```
Comment (didierofrivia) [inline config.toml:52]: This is added in line 3, you could remove it  from there if adding it under build

File: assets/js/greeter/Greeter.tsx (hunk 1 above)
Comment (damianpm) [reply assets/js/greeter/Greeter.tsx:4]: Yes, I'm keeping 2 spaces, I think is an issue with GH preview

File: package.json (hunk 3 above)
Comment (damianpm) [reply package.json:39]: They are similar, but last commit of `npm-run-all` was almost 3 years ago, last commit of `concurrently` 5 days ago.
Anyway, we can use any of them, no strong preference.

File: tsconfig.json [hunk 5]
```diff
@@ -1,14 +1,25 @@
...
+    "target": "es5",
+    "noImplicitAny": true,
+    "module": "es6",
+    "jsx": "react",
+    "allowJs": true,
+    "moduleResolution": "node",
+    "allowSyntheticDefaultImports": true,
+    "lib": [
+      "dom",
+      "es2018"
```
Comment (didierofrivia) [inline tsconfig.json:13]: add `"ESNext"` too

File: package.json (hunk 3 above)
Comment (didierofrivia) [reply package.json:39]: fair enough, let's keep it

**Pull request merged**